Rework HCP fingerprint generation (#12172)
hashicorp/packer: website/content/docs/hcp/index.mdx

You can set these additional environment variables to control how metadata is pushed to the registry.

> - `HCP_PACKER_BUILD_FINGERPRINT` - A unique identifier assigned to each iteration. HCP Packer uses this identifier to determine if metadata for a build on the registry is complete. By default, Packer will generate a new unique fingerprint for the template file on each invocation of `packer build`.

Author: I like the note and link to the iteration-fingerprinting section, regarding the second sentence for this doc I feel it's kinda paraphrasing the summary (and the extra section, which is probably where we can spend more time explaining the whats and hows for this variable). I think we can make this paragraph more concise, I'll push something for this.

> - `HCP_PACKER_BUILD_FINGERPRINT` - A unique identifier assigned to each build. HCP Packer uses this identifier to determine if metadata for a build on the registry is complete. By default, Packer will generate a new unique fingerprint for the template file on each new build. If you want to continue building an iteration, you will need to either set a unique fingerprint as HCP_PACKER_BUILD_FINGERPRINT before starting the build or you may use the generated one on subsequent builds.

Reviewer: > If you want to continue building an iteration, you will need to either set a unique fingerprint as HCP_PACKER_BUILD_FINGERPRINT before starting the build or you may use the generated one on subsequent builds.

I am finding this part a little confusing because `you may use the generated one on subsequent builds.` is also setting `HCP_PACKER_BUILD_FINGERPRINT`, right? 

```suggestion
- `HCP_PACKER_BUILD_FINGERPRINT` - A unique identifier assigned to each build. HCP Packer uses this identifier to determine if metadata for a build on the registry is complete. By default, Packer will generate a new unique fingerprint for the template file on each new build. If you want to continue building an iteration, you will need to set the HCP_PACKER_BUILD_FINGERPRINT with the iteration's fingerprint before starting the build.
```

Author: Yeah now that you mention it I tried to address two problems at once here, and I agree it can be confusing. Re-reading the comment there's a few more things that warrant a rephrasing here, I'll change that.
To your point, the difference between the two alternatives are:
1. you choose a unique pattern for your iteration before first building your template (i.e. first invocation of `packer build`)
2. you let Packer generate one for you, and will have to set it before running another build if you want to continue working on it (if there was an error or a `--only`/`--except` clause anywhere)

To avoid overcharging this bullet point, I'll probably add a paragraph for managing the fingerprint, so we can detail the alternatives and in which case you may want one over the other.
I also wonder if we should add this to this page or if we should create another one. I'll add this to that page for now, and we can discuss if that'd be better elsewhere.

Refer to [`hcp_packer_registry`](docs/templates/hcl_templates/blocks/build/hcp_packer_registry) for a full list of configuration arguments. Refer to [Custom Configuration](https://cloud.hashicorp.com/docs/packer/store-image-metadata/packer-template-configuration#custom-configuration) in the HCP Packer documentation for information and examples about how to customize image metadata.

### Iteration Fingerprinting

> Packer uses a unique fingerprint for tracking the complete of builds associated to an iteration. By default a fingerprint is automatically generated by Packer during each invocation of `packer build`. Unless a fingerprint is manually provided via the `HCP_PACKER_BUILD_FINGERPRINT` environment variable.

Author: Good catch! I'll fold this in and merge when tests pass
